### PR TITLE
[12.0][RM] Unnecessary dependence on hr and fieldservice_stock module

### PR DESCRIPTION
**Pull request by bodedra**
Issue Reference: https://github.com/OCA/field-service/issues/346

### CODE REVIEW

**Comment by bodedra** [comment]
CC @max3903

File: fieldservice_maintenance/models/fsm_order.py [hunk 1]
```diff
@@ -10,31 +10,3 @@ class FSMOrder(models.Model):
...
-    #                         equipment.maintenance_equipment_id.id,
-    #                     'category_id': equipment.category_id.id,
-    #                     'request_date': fields.Date.context_today(self),
-    #                     'maintenance_type': 'corrective',
-    #                     'maintenance_team_id': team_id,
-    #                     'schedule_date': order.request_early,
-    #                     'description': order.description
-    #                 })
-    #             order.request_id = request_id
-    #     return order
```
Comment (max3903) [inline fieldservice_maintenance/models/fsm_order.py]: @bodedra We need to uncomment this method to create the maintenance request.

Comment (bodedra) [reply fieldservice_maintenance/models/fsm_order.py]: @max3903 This method is already in comment. I just remove commented code.

If our goal to uncomment this method, then `hr` module is a valid as a dependency.

Comment (max3903) [reply fieldservice_maintenance/models/fsm_order.py]: We can remove the employee_id field and the dependency on hr.

Comment (bodedra) [reply fieldservice_maintenance/models/fsm_order.py]: Sure. Will do it.

File: fieldservice_maintenance/models/fsm_order.py [hunk 2]
```diff
@@ -11,30 +11,30 @@ class FSMOrder(models.Model):
...
+    @api.model
+    def create(self, vals):
+        # if FSM order with type maintenance is create then
+        # create maintenance request
+        order = super(FSMOrder, self).create(vals)
+        if order.type == 'maintenance':
```
Comment (max3903) [inline fieldservice_maintenance/models/fsm_order.py]: @bodedra The type is now a many2one field. We need to use the external id.

Comment (bodedra) [reply fieldservice_maintenance/models/fsm_order.py]: @max3903 Yes. I first uncomment lines and correct the logic.

**Comment by bodedra** [comment]
@max3903 Comment attended!

**Review by max3903 — APPROVED** [review]

**Comment by max3903** [comment]
@zbik2607 Can you review and provide your feedbacks?

**Comment by bodedra** [comment]
Are you available for review ? @brian10048 @hparfr

**Review by hparfr — COMMENTED** [review]
Why not make use of inheritence instead of create() in create() ?

**Comment by bodedra** [comment]
@hparfr IMO That's the another way of doing it but this will not harm. We are only create a maintenance request if FSM order type is maintenance.

**Comment by zbik2607** [comment]
if equipment.maintenance_equipment_id is not set, then creating maintenance.request probably doesn't make sense?

**Comment by bodedra** [comment]
> if equipment.maintenance_equipment_id is not set, then creating maintenance.request probably doesn't make sense?

Yes. Agree. I will improve this part because 'maintenance_equipment_id' field is not required.

Good catch.

**Comment by bodedra** [comment]
@zbik2607 Comment addressed!

**Comment by zbik2607** [comment]
I suggest deleting also dependency on fieldservice_stock - it is redundant too. 
Also replace lines in fsm_equipemt.py:

            'serial_no':
                vals['lot_id'] and
                self.env['stock.production.lot'].browse(vals['lot_id']).name
                or False

by:

            'serial_no':
                vals.get('lot_id', False) and
                self.env['stock.production.lot'].browse(vals['lot_id']).name

and remove view: fsm_equipment_form_view_model, move field model to fsm_equipment_form_view_maintenance

**Comment by zbik2607** [comment]
Sorry, for small fix  in updated my last answer:

"and remove view: fsm_equipment_form_view_model, move field model to fsm_equipment_form_view_maintenance"

**Comment by zbik2607** [comment]
Its OK :)

**Comment by bodedra** [comment]
@zbik2607 That's great. Thanks for your review. You can approve it.

**Comment by zbik2607** [comment]
Rather, it should be: vals.get ('lot_id', False)
instead of vals ['lot_id']
because lot_id may be absent

**Comment by bodedra** [comment]
@zbik2607 Comment addressed!

**Comment by zbik2607** [comment]
Everything seems to be correct now.

**Comment by bodedra** [comment]
> Everything seems to be correct now.

Sounds good. Thank you. You can give your official approval vote from the right hand side, file changes. :)

**Review by zbik2607 — APPROVED** [review]
Everything seems to be correct now.

**Comment by brian10048** [comment]
Do we know why @osi-scampbell commented the create method in #241?

@bodedra It may be worth improving fieldservice_repair as well? https://github.com/OCA/field-service/blob/ba82cadc01ecc35194447bcff595f27efd86ee45/fieldservice_repair/models/fsm_order.py#L15

**Comment by bodedra** [comment]
@brian10048 I don't the reason. @osi-scampbell would you please help us to here ?

Sure. Let's create a separate issue for improving fieldservice_repair module.

**Comment by max3903** [comment]
> Do we know why @osi-scampbell commented the create method in #241?

It was not working well enough.

> @bodedra It may be worth improving fieldservice_repair as well?
> 
> https://github.com/OCA/field-service/blob/ba82cadc01ecc35194447bcff595f27efd86ee45/fieldservice_repair/models/fsm_order.py#L15

Yes

**Comment by zbik2607** [comment]
I see other errors, but it's probably better to approve and create new issues. For example:  https://github.com/OCA/field-service/issues/370

**Review by hparfr — APPROVED** [review]

**Comment by max3903** [comment]
@bodedra Can you fix #370 as it is related to the same module?

**Comment by bodedra** [comment]
@max3903 suggested code from @zbik2607 is added.

**Comment by bodedra** [comment]
@max3903 This is ready for merge.

**Pull request merged**